types/block.go

```diff
@@ -409,7 +409,7 @@ func (h Header) ValidateBasic() error {
 		return fmt.Errorf("wrong EvidenceHash: %v", err)
 	}
 
-	if len(h.ProposerAddress) != crypto.AddressSize {
+	if len(h.ProposerAddress) > 0 && len(h.ProposerAddress) != crypto.AddressSize {
```

Comment:
why change?

Reply:
this was to keep inline with validateHash(). There it checks it is > 0 and the correct length. I can remove if youd like but not sure when this would be len(0) so might as well check

Reply:
but unless crypto.AddressSize is zero this already checks this

Reply:
ah yes, will remove